gazebo_ros/include/gazebo_ros/testing_utils.hpp

```diff
@@ -0,0 +1,155 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GAZEBO_ROS__TESTING_UTILS_HPP_
+#define GAZEBO_ROS__TESTING_UTILS_HPP_
+
+#include <unistd.h>
+#include <stdlib.h>
+
+#include <gtest/gtest.h>
+#include <rclcpp/rclcpp.hpp>
+
+#include <utility>
+#include <vector>
+#include <string>
+
+namespace gazebo_ros
+{
+
+
+/// Helper class to run gzserver in a seperate process and later terminate that process
+class GazeboProcess
+{
+public:
+  /// Start gzserver with a list of arugments
+  /// \note The path and --verbose are automaticaly added
+  explicit GazeboProcess(const std::vector<const char *> & args);
+
+  ~GazeboProcess();
+
+  /// Start gzserver with the arguments passed to constructuor
+  /// \return The result of fork(), either the pid of gazebo process or error if < 0
+  int run();
+
+  /// Terminate the child gzserve process
+  /// \return -1 if run() failed or has not been called,
+  int terminate();
+
+private:
+  /// Arguments to run gzserver with
+  std::vector<const char *> arguments;
+
+  // pid of gzserver
+  int pid_ = -1;
+};
+
+GazeboProcess::GazeboProcess(const std::vector<const char *> & args)
+{
+  arguments = {"/usr/bin/gzserver", "--verbose"};
+  arguments.insert(arguments.end(), args.begin(), args.end());
+  arguments.push_back(nullptr);
+}
+
+GazeboProcess::~GazeboProcess()
+{
+  terminate();
+}
+
+int GazeboProcess::run()
+{
+  // Fork process so gazebo can be run as child
+  pid_ = fork();
+
+  // Child process
+  if (0 == pid_) {
+    // Run gazebo with arguments
+    if (execvp("gzserver", const_cast<char **>(arguments.data()))) {
+      // Exec failed, cannot return (in seperate process), so just print errno
+      printf("gzserver failed with errno=%d", errno);
+      exit(1);
+    }
+  }
+
+  if (pid_ < 0) {
+    return errno;
+  }
+
+  // Parent process, return pid of child (or error produced by fork())
+  return pid_;
+}
+
+int GazeboProcess::terminate()
+{
+  // Return -1
+  if (pid_ < 0) {
+    return ECHILD;
+  }
+
+  // Kill gazebo (simulating ^C command)
+  if (kill(pid_, SIGINT)) {
+    return errno;
+  }
+
+  // Wait for gazebo to terminate
+  if (waitpid(pid_, nullptr, 0) < 0) {
+    return errno;
+  }
+
+  return 0;
+}
+
+
+/// Helper function to get the next message on a ROS topic with a timeout
+/// \param node Pointer to a node to use to subscribe to the topic
+/// \param topic Topic to subscribe to for message
+/// \param timeout Maximum time to wait for message
+/// \tparam T Message type to get
+/// \return Shared pointer to new message, or nullptr if none received before timeout
+template<typename T>
+typename T::SharedPtr
+get_message_or_timeout(
```

Comment:
This function is not being used, did you mean to use it?

Reply:
Used for plugin test in 815fc64